miniml/compiler/hello.ml

```diff
@@ -269,5 +269,20 @@ let rec stack_overflow () = 1 + stack_overflow ()
 let () = run_and_print_exn stack_overflow
 let () = run_and_print_exn (fun () -> div 1 0)
 
+
+let () = print "\nFunctors:\n"
+
+module X(A : sig val x : int end) = struct
+  let x = 2 * A.x
+end
+
+module Y = X(struct let x = 21 end)
+module Z = X(struct let x = 12 end)
```

Comment:
Nitpick: I think it could be useful to have a test where we call `F(X)` several times, checking that while the effects in `F` are duplicated, the effects in the evaluation of `X` (previously defined as `module X = struct ... end`) are not, that its components are reduced to values and then shared between functor applications.

Reply:
(Also nitpick: I am more familiar with naming conventions where functors are called `F, G`, functor arguments are `M` or `X, Y, Z`, and modules are something like `M` or `A, B, C`.)